From 580cc23fd90c15259fbb999a8e8d82a93abdcc31 Mon Sep 17 00:00:00 2001
From: Khem Raj <raj.khem@gmail.com>
Date: Tue, 26 Oct 2021 11:03:25 -0700
Subject: [PATCH] Fix build on mips

Fixes
Source/WTF/wtf/Atomics.cpp:63:9: error: definition of builtin function '__sync_add_and_fetch_8'
int64_t __sync_add_and_fetch_8(int64_t volatile* addend, int64_t value)
Source/WTF/wtf/Atomics.cpp:68:9: error: definition of builtin function '__sync_sub_and_fetch_8'
int64_t __sync_sub_and_fetch_8(int64_t volatile* addend, int64_t value)

Upstream-Status: Pending

Signed-off-by: Khem Raj <raj.khem@gmail.com>
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 Source/WTF/wtf/Platform.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/WTF/wtf/Platform.h b/Source/WTF/wtf/Platform.h
index 89d988231..3746c957a 100644
--- a/Source/WTF/wtf/Platform.h
+++ b/Source/WTF/wtf/Platform.h
@@ -88,6 +88,7 @@
 #else
 #define WTF_CPU_MIPS 1
 #define WTF_MIPS_ARCH __mips
+#define __GCC_HAVE_SYNC_COMPARE_AND_SWAP_8 1
 #endif
 #if defined(__MIPSEB__)
 #define WTF_CPU_BIG_ENDIAN 1
